Relax dependency on bundler

mkllnk opened the pull request:
Because of recently published security vulnerabilities I looked into the used bundler version here. Dependabot couldn't determine the version.

Bundler doesn't manage itself with the Gemfile.lock. You can't run
`bundle update bundler`. So the version specification here doesn't mean
anything. The Gemfile.lock contains the BUNDLED WITH line which reveals
the used version though.

I also looked into my installed gems. Most have a Gemfile. But less than
half mention bundler in the gemspec. So we could go further and remove
bundler from the spec completely...

But for now I just removed the version number.

mkllnk committed:
Relax dependency on bundler

Bundler doesn't manage itself with the Gemfile.lock. You can't run
`bundle update bundler`. So the version specification here doesn't mean
anything. The Gemfile.lock contains the BUNDLED WITH line which reveals
the used version though.

I also looked into my installed gems. Most have a Gemfile. But less than
half mention bundler in the gemspec. So we could go further and remove
bundler from the spec completely...

dacook commented:
Hi @mkllnk, I agree that we don't need an explicit dependency on bundler and it can be removed entirely.

However, the vulnerability described here affects the version you 'bundled with'. It seems only the latest version (2.2.18 released yesterday) is safe. So we need to:
* update our locally installed gem (`gem install bundler`), and
* update `Gemfile.lock` in each repo to use it (`bundle update --bundler`)

(https://bundler.io/guides/bundler_2_upgrade.html)

While you're at it, would you be able to update this repo?

mkllnk commented:
Yes, we need to update but we need Rails 5.2 for that. Rails before that has a dependency on bundler < 2.0.

mkllnk committed:
Remove bundler from dev dependency list

We don't really need to include it here because the Gemfile.lock
defining the version already.

dacook reviewed (approved):
Thanks Maikel
